modules/src/crypto.py

```diff
@@ -0,0 +1,54 @@
+import requests
+
+from templates.text import TextTemplate
+
+
+def process(entry, entities):
+    output = {}
+
+    class BittrexAPIException(Exception):
+        error_message = 'I could not obtain the information.'
+        error_message += '\nPlease try again later!'
+        output['error_msg'] = error_message
+        output['success'] = False
+
+    try:
+        market = entities['market'][0]['value'].upper()
+        if 'to' in market:
+            market = market.replace('to', '-').replace(' ', '')
+
+        r = requests.get('https://bittrex.com/api/v1.1/public/getticker?market=' + market)
+        data = r.json()
+
+        if not data['success']:
+            market += '-' + market[:market.find('-')]
+            market = market[market.find('-')+1:]
+            r = requests.get('https://bittrex.com/api/v1.1/public/getticker?market=' + market)
+            data = r.json()
+
+        if not data['success']:
+            raise BittrexAPIException(data)
```

Comment:
[Here](https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python)  very thoroughly described raising an exception

Reply:
Thank you for the link and the guidance!

As for the question, I was following the documentation for errors and exceptions. I was planning to add a user-defined exception after I had other types of exceptions for other API calls.